### PR TITLE
Use catalog_name if defined

### PR DESCRIPTION
**Pull request by bchaillou-ovh**
# Description

This PR is a request from Webhosting Infra team:
- Add catalog_name on order_cart (for OptionsPlan). Unit tests updated.

## Type of change

- [X] Improvement (improve existing resource(s) or datasource(s))
- [X] Documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

It is difficult to test by doing an acceptance test. You should contact Yoann Normand from the Webhosting Infra team who took care to make a manually tested.

### CODE REVIEW

**Review by scraly — CHANGES_REQUESTED** [review]
Can you add a working example with a "catalog_name" in the documentation?

**Comment by bchaillou-ovh** [comment]
Example updated with catalog_name.

**Review by scraly — CHANGES_REQUESTED** [review]
Acceptance tests fails 

```
$ make testacc TESTARGS="-run TestAccDataSourceOrderCartIpLoadbalancingOptionsPlan_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run TestAccDataSourceOrderCartIpLoadbalancingOptionsPlan_basic -timeout 600m -p 10
?   	github.com/ovh/terraform-provider-ovh	[no test files]
?   	github.com/ovh/terraform-provider-ovh/ovh/helpers	[no test files]
=== RUN   TestAccDataSourceOrderCartIpLoadbalancingOptionsPlan_basic
    data_order_cart_product_options_plan_test.go:64: Step 1/1 error: Error running pre-apply refresh: exit status 1

        Error: Error calling Get /order/cart/xxxxxxxxxx/ipLoadbalancing/options?planCode=iplb-mutu&catalogName=iplb_private_beta:
        	 "OVHcloud API error (status code 400): Client::BadRequest: \"Cart needs to be assigned to specify a catalog name\" (X-OVH-Query-Id: EU.ext-xxxxxxxxxxxxxxxxxxxxxxx)"

          with data.ovh_order_cart_product_options_plan.plan,
          on terraform_plugin_test.tf line 8, in data "ovh_order_cart_product_options_plan" "plan":
           8: data "ovh_order_cart_product_options_plan" "plan" {

--- FAIL: TestAccDataSourceOrderCartIpLoadbalancingOptionsPlan_basic (26.94s)
FAIL
FAIL	github.com/ovh/terraform-provider-ovh/ovh	27.433s
testing: warning: no tests to run
PASS
ok  	github.com/ovh/terraform-provider-ovh/ovh/helpers/hashcode	0.260s [no tests to run]
FAIL
make: *** [testacc] Error 1
```

and

```
$ make testacc TESTARGS="-run TestAccDataSourceOrderCartIpLoadbalancingPlan_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run TestAccDataSourceOrderCartIpLoadbalancingPlan_basic -timeout 600m -p 10
?   	github.com/ovh/terraform-provider-ovh	[no test files]
?   	github.com/ovh/terraform-provider-ovh/ovh/helpers	[no test files]
=== RUN   TestAccDataSourceOrderCartIpLoadbalancingPlan_basic
    data_order_cart_product_plan_test.go:44: Step 1/1 error: Error running pre-apply refresh: exit status 1

        Error: Error calling Get /order/cart/xxxxxxx/ipLoadbalancing?catalogName=iplb_private_beta:
        	 "OVHcloud API error (status code 400): Client::BadRequest: \"Cart needs to be assigned to specify a catalog name\" (X-OVH-Query-Id: EU.ext-xxxxxxxxxxxx)"

          with data.ovh_order_cart_product_plan.plan,
          on terraform_plugin_test.tf line 8, in data "ovh_order_cart_product_plan" "plan":
           8: data "ovh_order_cart_product_plan" "plan" {

--- FAIL: TestAccDataSourceOrderCartIpLoadbalancingPlan_basic (17.42s)
FAIL
FAIL	github.com/ovh/terraform-provider-ovh/ovh	17.840s
testing: warning: no tests to run
PASS
ok  	github.com/ovh/terraform-provider-ovh/ovh/helpers/hashcode	0.241s [no tests to run]
FAIL
make: *** [testacc] Error 1
```

File: website/docs/d/order_cart_product_options_plan.html.markdown [hunk 1]
```diff
@@ -20,13 +20,14 @@ data "ovh_order_cart_product_options_plan" "plan" {
...
   options_plan_code = "vrack"
+  catalog_name      = "iplb_private_beta"
 }
 ```
 
 ## Argument Reference
 
 * `cart_id` - (Required) Cart identifier
-* `catalog_name` - Catalog name
+* `catalog_name` - Catalog name (Internal API - restricted to OVHcloud internal employees)
```
Comment (amstuta) [inline website/docs/d/order_cart_product_options_plan.html.markdown:30]: It is not acceptable to add parameters that are restricted to OVHcloud employees on the public provider. You should check internally with the team responsible for these routes if it's possible to make this parameter public and then we will check this PR.

**Review by amstuta — CHANGES_REQUESTED** [review]

**Comment by amstuta** [comment]
Closing this request for inactivity, don't hesitate to re-open it if `catalog_name` field goes public.

**Pull request closed without merging**